flac.md

```diff
@@ -654,6 +654,57 @@ The reason for this limit is to ensure that decoders can use 32-bit integers whe
+
+## Matroska mapping
+
+The Matroska container format is defined in [@!I-D.ietf-cellar-matroska]. The codec ID (EBML path `\Segment\Tracks\TrackEntry\CodecID`) assigned to signal tracks carrying FLAC data is `A_FLAC` in ASCII. All FLAC data before the first audio frame (i.e. the `fLaC` ASCII signature and all metadata blocks) is stored as CodecPrivate data (EBML path `\Segment\Tracks\TrackEntry\CodecPrivate`).
+
+Each FLAC frame (including all of its subframes) is treated as a single frame in the Matroska context.
+
+In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this should be dealt with a by finishing the current Matroska segment and starting a new one with the new parameters.
+
+## ISO Base Media File Format (MP4) mapping
+
+The full encapsulation definition of FLAC audio in MP4 files was deemed too extensive to include in this document. A definition document can be found at https://github.com/xiph/flac/blob/master/doc/isoflac.txt This document is summarized here.
+
+The sample entry code is 'fLaC'. No object type indication is registered. The channelcount and samplesize fields in the sample entry follow from the values as found in the FLAC stream. The samplerate field can be different, because FLAC can carry audio with much higher samplerates than can be coded for in the sample entry. When possible, the samplerate field should contain the sample rate as found in the FLAC stream, shifted left by 16 bits to get the 16.16 fixed point representation of the samplerate field. When the FLAC stream contains a sample rate higher than can be coded, the samplerate field contains the greatest expressible regular division of the sample rate, e.g. 48000 for sample rates of 96kHz and 192kHz or 44100 for a sample rate of 88200Hz. When the FLAC stream contain audio with an unusual samplerate that has no regular division, the maximum value of 65535.0 Hz is used. As FLAC streams with a high sample rate are common, a parser or decoder MUST read the value from the FLAC streaminfo metadata block or a frame header to determine the actual sample rate. The sample entry contains one 'FLAC specific box' with code 'dfLa'.
+
+The FLAC specific box has version number 0 and contains the metadata blocks of the FLAC stream. Contrary to the CodecPrivate data in the Matroska mapping, this excludes the `fLaC` ASCII signature.
+
+In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this should be dealt with **TODO: Someone with knowledge of this should chime in. This is NOT handled in the mapping document**
```

Comment:
```suggestion
In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this SHOULD be dealt in a MP4 generic manner i.e. with several `stsd` atoms and different sample-description-index values in the `stsc` atom.
```

Reply:
I see now I used should for similar phrases for Matroska and Ogg, perhaps MUST is better?

Reply:
Reluctant with MUST, e.g in case there is a new method invented in MP4 in the future, FLAC does not care actually, it is just out of the scope. So replacing "i.e." by "e.g." (meaning for me that FLAC is not specific to how it is done in MP4).:

```suggestion
In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this MUST be dealt in a MP4 generic manner e.g. with several `stsd` atoms and different sample-description-index values in the `stsc` atom.
```